### DISCUSSION TITLE
Hide mkdocs material webpage behind a login system

### DISCUSSION THREAD

**Discussion by zerenxyz** [squidfunk/mkdocs-material#5050] [Q&A] [answered]
Is it possible somehow to hide the mkdocs material insiders install with github pages behind a login system of some sort so it is a private wiki/docs. As the purpose I am trying to use it for within the company will have sensitive data on it and we need to have it secure.

I have looked into mkdocs encrypt plugin which works sort of but the user needs to add a password for each page etc, I am looking for more a solution if possible where you login with a given name/password (doesnt need multi-account, one account is fine) and that will give you read access to the website.

**Comment by squidfunk** [comment, marked as answer]
Of course! You can host the static site that is generated by MkDocs and Material for MkDocs with any service or technology that allows for web hosting. For example, you could use [nginx](https://www.nginx.com/) and add [basic authentication](https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/) to protect your site, or use IP whitelisting, OAuth (with JWT), ... basically any authentication method. It's beyond the scope of Material for MkDocs, though.

I would recommend to talk to your ops team. Related: #3854

**Reply by squidfunk** [reply]
Note that GitHub pages doesn't allow to secure content – you would need to choose another solution.

**Reply by zerenxyz** [reply]
Ah you mean the static site that is generated using mkdocs build? Does linking between pages and all work within that static build then? The nginx and basic authentication seems like a nice easy solution! From your next reply I assume that is not possible to achieve within github pages, do you have any recommendations on where to host the static site by any chance? Locally, and or for example digital ocean?

**Reply by squidfunk** [reply]
You can follow _any_ guide on [static site hosting with authentication](https://www.google.com/search?q=static+site+hosting+basic+auth).

**Reply by zerenxyz** [reply]
I was able to make use of basic authentication for this and it seems to work! Thank you for the information!
